### PR TITLE
Setup CI for Serverless 1.22

### PR DESCRIPTION
**Pull request by skonto**
As per title. Copied from main. `make jobs ci-operator-config` introduces periodics not sure if we need it.
/cc @nak3 @mgencur @matzew

### CODE REVIEW

File: ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml [hunk 1]
```diff
@@ -0,0 +1,1068 @@
+periodics:
```
Comment (skonto) [inline ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml:1]: Do we need it?

Comment (pierDipi) [reply ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml:1]: We don't run them for other versions, so maybe not?

Comment (skonto) [reply ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml:1]: I can remove it but not sure if the config jobs will pass.

Comment (skonto) [reply ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml]: config does not pass if removed:
```
diff -Naupr .//ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml /tmp/tmp.pUU7YGf2Rm/ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml
--- .//ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml	1970-01-01 00:00:00.000000000 +0000
+++ /tmp/tmp.pUU7YGf2Rm/ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml	2022-04-05 19:34:10.098611968 +0000
```

**Comment by skonto** [comment]
@nak3 gentle ping.

**Review by nak3 — CHANGES_REQUESTED** [review]
Please fix `ci/prow/generated-config`. It just needs to run `$ make jobs`.

File: ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml (hunk 1 above)
Comment (mgencur) [reply ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml]: We don't want the periodic jobs. We only want them for `main` but not for this release branch. They're defined in the ci-operator config (e.g. [here](https://github.com/openshift/release/pull/27571/files#diff-46b7739dde0304cfcae92fbd1cbcc525499630a00471be3b0fb39ce7be05d34dR167)) - all the checks with `cron` definition in them (including UI tests)

File: ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22__4.10.yaml [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+        name: openshift-knative-operator
+      from: serverless-operator-src
+      resources:
+        limits:
+          memory: 6Gi
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    workflow: generic-claim
+- as: e2e-aws-ocp-410-continuous
```
Comment (mgencur) [inline ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22__4.10.yaml:167]: Let's remove this periodic run.

File: ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22__4.10.yaml [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+        name: openshift-knative-operator
+      from: serverless-operator-src
+      resources:
+        limits:
+          memory: 6Gi
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    workflow: generic-claim
+- as: upgrade-tests-aws-ocp-410-continuous
```
Comment (mgencur) [inline ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22__4.10.yaml:195]: Let's remove this periodic run.

File: ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22__4.10.yaml [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+        name: openshift-knative-operator
+      from: serverless-operator-src
+      resources:
+        limits:
+          memory: 6Gi
+        requests:
+          cpu: "2"
+          memory: 4Gi
+    workflow: generic-claim
+- as: e2e-mesh-ocp-410-continuous
```
Comment (mgencur) [inline ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22__4.10.yaml:252]: Let's remove this periodic run as well.

File: ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22__4.10.yaml [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+        name: openshift-knative-operator
+      from: serverless-operator-src
+      resources:
+        limits:
+          memory: 6Gi
+        requests:
+          cpu: "2"
+          memory: 4Gi
+    workflow: generic-claim
+- as: ui-tests
```
Comment (mgencur) [inline ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22__4.10.yaml:309]: Let's remove this periodic run as well.

**Review by mgencur — COMMENTED** [review]
In general, I think it would have been easier to copy the setup from one of the previous release branches and not from `main`.
For the main branch we want to have a few checks in pre-submit phase and several checks periodic. Especially, we only run the last OCP version in pre-submits and the rest periodically.
For the release branch we don't want any periodic runs but we want to have all checks in pre-submits - for all OCP version (probably all OCP 4.6 - 4.10), or we can decide to reduce it this time and have only OCP 4.10 + OCP 4.6 to save resources. The QE team will test the other versions.
One last note: If copying the setup from previous release branches then 1.20 is a better candidate because 1.21 was already a bit sub-optimal because it doesn't use cluster pools for some versions.

**Comment by skonto** [comment]
I wouldnt say easier because I would have to update the image numbers. The last time I did that way and it was also a bit of pain.

**Comment by mgencur** [comment]
> It didnt seem smooth (to copy from 1.20+) because I would have to update the image numbers. The last time I did that way and it was also a bit of pain. Is there anything else needed beyond removing the periodics?

Yeah. It's not smooth either way.
For this this release branch we don't need periodic but we need to run the checks in pre-submits. For previous release branches we were running all checks in pre-submits for all OCP versions. But this list of CI checks and OCP versions is constantly growing and I think it's not feasible to do it again in the same way.
Personally, I'd reduce it in this way:
Run all of the checks below and run them only on the latest and oldest supported OCP- 4.10 and 4.6 :
```
- as: operator-e2e-aws-ocp-410
- as: upstream-e2e-aws-ocp-410
- as: upgrade-tests-aws-ocp-410
- as: upstream-e2e-mesh-aws-ocp-410
```
With this setup we'll have 8 pre-submit checks (better than 20!)

File: ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml (hunk 1 above)
Comment (skonto) [reply ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-release-1.22-periodics.yaml]: So runnign again `make....` without periodics this part is not generated anymore.Makes sense.

**Comment by skonto** [comment]
@mgencur
a)  I copied the tests section from 1.20 for all serving configs, removed all periodics.
b) in pre-submits kept the 8 tests you mentioned plus images per ocp version and unit tests.
Pls review.

**Comment by skonto** [comment]
@mgencur limiting the pre-submit tests breaks several validation jobs here. Running `make jobs ci-operator-config` will fill in all jobs for all versions I removed: 4.7,4.8,4.9. Any idea how to avoid this?

**Comment by mgencur** [comment]
/lgtm
/hold

Feel free to unhold when the pj-rehearse tests pass or if you want to skip them.

**Comment by skonto** [comment]
timed out during removal (/cc @pierDipi ):

```
done
13:40:49.617 INFO:    Removing KnativeEventing CR
knativeeventing.operator.knative.dev "knative-eventing" deleted
13:41:08.994 INFO:    Removing KnativeKafka CR
knativekafka.operator.serverless.openshift.io "knative-kafka" deleted
13:41:11.169 INFO:    Ensure no knative eventing or knative kafka pods running
13:41:11.171 DEBUG:   [[ $(oc get pods -n knative-eventing --field-selector=status.phase!=Succeeded -o jsonpath='{.items}') != '[]' ]] : Waiting until non-zero (max 600 sec.)
.........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
13:52:11.722 ERROR:   Time out of 600 exceeded
```

**Comment by matzew** [comment]
> Error from server (NotFound): namespaces "kafka" not found
that is ... interesting?!

**Comment by mgencur** [comment]
@matzew Right. I was just going to say that there's an actual problem with kafka-controller-post-install.
It doesn't seem to have anything in common with this PR. I think this PR can be merged and the actual problem fixed in the product later - with a separate PR to serverless-operator or elsewhere.

**Comment by mgencur** [comment]
> > Error from server (NotFound): namespaces "kafka" not found
> that is ... interesting?!

I can see this in the log: 
```
The connection to the server api.ci-ocp-4-10-amd64-aws-us-east-1-mjzfj.hive.aws.ci.openshift.org:6443 was refused - did you specify the right host or port?
````
And it happens at the beginning of installing Strimzi so I guess it failed just when the script was trying to create the kafka namespace.

**Comment by matzew** [comment]
@mgencur the post install has been addressed

**Comment by skonto** [comment]
It seems the CR issue still exists looking at the last run.

**Comment by matzew** [comment]
> It seems the CR issue still exists looking at the last run.

image cache issue?

Doing it again:

/retest

**Comment by pierDipi** [comment]
(I will take a look directly in SO and locally)

**Pull request merged**